doc/source/whatsnew/v1.5.0.rst

```diff
@@ -267,6 +267,7 @@ Categorical
 ^^^^^^^^^^^
 - Bug in :meth:`Categorical.view` not accepting integer dtypes (:issue:`25464`)
 - Bug in :meth:`CategoricalIndex.union` when the index's categories are integer-dtype and the index contains ``NaN`` values incorrectly raising instead of casting to ``float64`` (:issue:`45362`)
+- Bug in :meth:`CategoricalIndex._concat` when concatenating categorical indexes that have the same categories returning indexes in improper order; this method can be called in :meth:`concat` (:issue:`44099`)
```

Comment:
can you update the note here; this should be about the user facing case in the OP

Reply:
Thanks! I added description line at `reshaping` section that refer to `concat` func in this commit https://github.com/pandas-dev/pandas/pull/46019/commits/3425b854ca9f3c9370ad1cfb36f03d3ba9b58000

Is this the correct fix that you pointed out?

Reply:
@jreback Hi! Thanks for all your reviews.

I would like to request a review from you again.
I finally change [doc/source/whatsnew/v1.5.0.rst](https://github.com/pandas-dev/pandas/pull/46019/files/aacc75c6807b882f6142007f8802d92f80815a74#diff-c84bc2c71f43444548da8b2eb7b2f34efb50dac71eeddabbfcdd8ae7d7badc4b) in this commit([0e62a36](https://github.com/pandas-dev/pandas/pull/46019/commits/0e62a362b761f5069980f29aef0e40a5f024225b))